app/javascript/src/Applications/ApplicationForm.jsx

```diff
@@ -0,0 +1,161 @@
+// @flow
+
+import React, { useState, useEffect } from 'react'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import './applicationForm.scss'
+
+type ApplicationPlan = {
+  id: number,
+  name: string,
+}
+
+type ServicePlan = {
+  id: number,
+  name: string,
+  default: boolean
+}
+
+type Props = {
+  plans: ApplicationPlan[],
+  servicesContracted: number[],
+  relationServiceAndServicePlans: {[number]: ServicePlan[]},
+  relationPlansServices: {[number]: number},
+  servicePlanContractedForService: {[number]: ServicePlan},
+  servicePlansAllowed: boolean
+}
+
+const ApplicationForm = ({
+  plans, servicesContracted, relationServiceAndServicePlans,
+  relationPlansServices, servicePlanContractedForService, servicePlansAllowed
+}: Props) => {
+  const [selectedPlan, setSelectedPlan] = useState(plans[0])
+  const [term, setTerm] = useState(selectedPlan.name)
+
+  function checkSelectedPlan () {
+    const serviceId = getServiceIdOfPlanId(selectedPlan.id)
+    const servicePlans = getServicePlansForService(serviceId)
+
+    enableForm()
+
+    if (servicesContracted.indexOf(serviceId) > -1) {
+      const { id, name } = getContractedServicePlanForService(serviceId)
+      $('#cinstance_service_plan_id').html(`<option value="${id}"> ${name} </option>`)
```

Comment:
here we are relying on a library which is not included in our file. Also, we need to ditch the usage of this precise one ;) . If you are planning to remove it later in a different PR, at least include jQuery in this file. The usage of `$` applies to the whole file

Reply:
In progress